### PR TITLE
add build support for elementary OS, also consistent spaces vs tabs a…

### PR DESCRIPTION
**Pull request by rogusdev**
…nd removing trailing spaces

### CODE REVIEW

File: build/common_functions [hunk 1]
```diff
@@ -107,7 +107,7 @@ function GetOSVersion {
         os_RELEASE=$(lsb_release -r -s)
         os_UPDATE=""
         os_PACKAGE="rpm"
-        if [[ "Debian,Ubuntu,LinuxMint" =~ $os_VENDOR ]]; then
+        if [[ "Debian,Ubuntu,LinuxMint,elementary" =~ $os_VENDOR ]]; then
```
Comment (rogusdev) [inline build/common_functions:110]: add the new OS vendor

File: src/configure [hunk 2]
```diff
@@ -35,37 +35,54 @@ if [ "$os_VENDOR" == "Ubuntu" ] && [ "$#" -gt 2 ]; then
 fi
 
 
-if [ "$os_VENDOR" == "Ubuntu" ] || [ "$os_VENDOR" == "LinuxMint" ]; then
-    ubuntu_ver=${os_RELEASE:0:2}
-    print_title "Build RDM on $os_VENDOR: $ubuntu_ver"         
+if [ "$os_VENDOR" == "Ubuntu" ] || [ "$os_VENDOR" == "LinuxMint" ] || [ "$os_VENDOR" == "elementary" ]; then
```
Comment (rogusdev) [inline src/configure:38]: include the new vendor

File: src/configure [hunk 3]
```diff
@@ -35,37 +35,54 @@ if [ "$os_VENDOR" == "Ubuntu" ] && [ "$#" -gt 2 ]; then
...
+
+    if [ "$os_VENDOR" == "elementary" ]; then
+        ubuntu_ver=${os_RELEASE:0:3}
```
Comment (rogusdev) [inline src/configure:41]: elementary version has a "0.4.1" format, not based off of ubuntu versions

File: src/configure [hunk 4]
```diff
@@ -35,37 +35,54 @@ if [ "$os_VENDOR" == "Ubuntu" ] && [ "$#" -gt 2 ]; then
...
     if [ "$os_VENDOR" == "Ubuntu" ] && [ "$ubuntu_ver" == "12" ]; then
         echo "RedisDesktopManager doesn't support Ubuntu 12 since 0.8.8 release."
         exit 1
     fi
-    
-    if [ "$os_VENDOR" == "LinuxMint" ] && [ "$ubuntu_ver" -lt 17 ]; then
-        echo "RedisDesktopManager only supports linux mint >=17."
-        exit 1
+
+    if [ "$os_VENDOR" == "LinuxMint" ]; then
```
Comment (rogusdev) [inline src/configure:53]: Consolidate LinuxMint version conversion in the same style as used by elementary below

**Comment by rogusdev** [comment]
All the other changes are whitespace: tabs into spaces or removing trailing whitespace

**Comment by rogusdev** [comment]
It does also occur to me that maybe we could do something with the upstream flags on lsb_release, when available:
```
$ lsb_release -uc -s  # upstream codename
xenial
$ lsb_release -ur -s
16.04
$ lsb_release -ui -s
Ubuntu
```

**Comment by rogusdev** [comment]
Oh and this is elementary, an Ubuntu derivative like LinuxMint:
https://elementary.io/

**Comment by uglide** [comment]
Looks good to me. If you can implement `lsb_release -ur -s` with fallback to `lsb_release -r -s` it would be awesome! In that case, we can get rid of ugly `if` mappings for Linux Mint and Elementary

**Pull request merged**